content/guides/weird_characters.adoc

```diff
@@ -0,0 +1,846 @@
+= The Weird and Wonderful Characters of Clojure
+James Hughes
+2017-05-27
+:type: guides
+:toc: macro
+
+ifdef::env-github,env-browser[:outfilesuffix: .adoc]
+
+[]
+====
+This is a reference collection of characters used in Clojure that are difficult to "google".
+Descriptions are sourced from various blogs, https://stackoverflow.com[StackOverflow],
+http://en.wikibooks.org/wiki/Learning_Clojure[Learning Clojure], and the
+http://clojure.org/documentation[official Clojure docs] -- sources attributed
+where necessary. Sections are not in any particular order, but related items
+are grouped for ease. This guide is a copy of http://twitter.com/kouphax[James Hughes]
+brilliant https://yobriefca.se/blog/2014/05/19/the-weird-and-wonderful-characters-of-clojure/[blog post] of the same name.
+====
+[[dispatch]]
+== `#` - Dispatch macro
+
+You'll see this macro character beside another e.g. `\#(` or `#"`.
+// " Comment needed for emacs to behave.
+This topic will act as a bit of a preamble before looking at your specific case.
+
+`#` is a reader macro that tells the Clojure reader (the thing that takes a
+file of Clojure code and parses it for
+consumption by the compiler) to go and look at another *read table*
+for the definition of the next character - in essence this allows
+extending default reader behaviour.
+
+Clojure doesn't provide support for creating reader macros, but it is possible
+through http://briancarper.net/blog/449/[a bit of hackery].
+
+If you se `#` *at the end* of a symbol, then it is used to automatically
+generate a new symbol. This is useful inside macros to keep macro specifics
+from leaking into the userspace. A regular `let` will fail in a macro definition
+
+[source,clojure]
+----
+user=> (defmacro m [] `(let [x 1] x))
+#'user/m
+user=> (m)
+CompilerException java.lang.RuntimeException: Can't let qualified name: user/x, compiling:(NO_SOURCE_PATH:1)
+----
+
+Instead you need to append `#` to the end of the variable name and let Clojure
+generate a unique symbol for it:
+
+[source, clojure]
+----
+user=> (defmacro m [] `(let [x# 1] x#))
+#'user/m
+user=> (m)
+1
+user=>
+----
+
+If we expand this macro, we can see the `gensym` 'd name:
+
+[source, clojure]
+----
+user=> (macroexpand '(m))
+(let* [x__681__auto__ 1] x__681__auto__)
+----
+
+Another place you'll see the `#` is in
+<<xref/../../reference/reader#tagged_literals,tagged literals>>
+Most commonly you'll see this use in https://github.com/edn-format/edn[EDN]
+(extensible data notation - a rich data fromat that can be used in Clojure)
+and in ClojureScript (`#js`). See  <<xref/../weird_characters#tagged_literals,`#inst`, `#uuid`, or `#js`>> for more info.
+
+* <<xref/../../reference/reader#tagged_literals,Clojure Documentation Reader>>
+* http://briancarper.net/blog/449/[Clojure Reader Macros]
+* http://clojuredocs.org/clojure_core/clojure.core/gensym[ClojureDocs - gensyms]
+
+== `#{` - Set macro
```

Comment:
For all these with paired ending characters, I think it would be less confusing to include the end character. Like `#{` ... `}`. I would also remove "macro" here.